plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -698,7 +700,7 @@ public void dropTable(ConnectorSession session, ConnectorTableHandle tableHandle
     @Override
     public ConnectorOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, Optional<ConnectorTableLayout> layout, RetryMode retryMode)
     {
-        if (retryMode != NO_RETRIES) {
+        if (retryMode != NO_RETRIES && isNonTransactionalInsert(session)) {
```

Comment:
This looks wrong. I think you need to delegate decision "do we support retries" to jdbcClient now. Otherwise you will allow enabling retries even for connectors which pass `false` as `supportsFTE` when jdbc client is created.

Also I think it makes sense to split check if retries are generally supported and if they are used with non-transactional inserts. Sth like:
```java
        if (retryMode != NO_RETRIES && jdbcClient.supportsFaultTolerantExecution()) {
            throw new TrinoException(NOT_SUPPORTED, "This connector does not support query retries");
        }

        if (retryMode != NO_RETRIES && isNonTransactionalInsert()) {
            throw new TrinoException("Query retries are not compatible with non transactional inserts");
        }

```

Reply:
good catch, the change to this condition is one of the first changes i made for this PR